node/src/components/block_proposer.rs

```diff
@@ -86,28 +78,30 @@ impl Display for Event {
 }
 
 type DeployCollection = HashMap<DeployHash, DeployHeader>;
-pub type ProtoBlockCollection = HashMap<ProtoBlockHash, DeployCollection>;
+type FinalizationQueue = HashMap<u64, Vec<DeployHash>>;
+type RequestQueue = HashMap<u64, Vec<ListForInclusionRequest>>;
 
 pub(crate) trait ReactorEventT: From<Event> + From<StorageRequest> + Send + 'static {}
 
 impl<REv> ReactorEventT for REv where REv: From<Event> + From<StorageRequest> + Send + 'static {}
 
 /// Stores the internal state of the BlockProposer.
-#[derive(DataSize, Default, Debug, Clone, PartialEq)]
+#[derive(DataSize, Default, Debug)]
 pub(crate) struct BlockProposerState {
     pending: DeployCollection,
-    proposed: ProtoBlockCollection,
-    finalized: ProtoBlockCollection,
+    finalized_deploys: DeployCollection,
+    next_finalized: u64,
```

Comment:
Can we add a comment here? Or maybe change the name to `next_expected_finalized_height`? Although, that's a long name.